client/src/layout/Mention.js

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect } from "react";
+import {
+  Typography,
+  Card,
+  CardContent,
+  CardMedia,
+  Box,
+} from "@material-ui/core";
+import { makeStyles } from "@material-ui/core/styles";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faSmile } from "@fortawesome/free-regular-svg-icons";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    backgroundColor: theme.palette.common.white,
+    display: "flex",
+    alignItems: "center",
+    marginRight: theme.spacing(3),
+    marginTop: theme.spacing(3),
+    width: "80%",
+    height: "10em",
+    borderRadius: theme.shape.borderRadius * 2,
+    boxShadow: "0px 0px 2px 0px rgba(0,0,0,0.3)",
+    "&:hover": {
+      boxShadow: "0px 0px 4px 1px rgba(0,0,0,0.3)",
+    },
+  },
+  media: {
+    minHeight: "8em",
+    minWidth: "8em",
+    marginLeft: "1em",
+    border: "1px solid black",
+  },
+  content: {
+    padding: theme.spacing(3),
+    display: "flex",
+    flexDirection: "column",
+    textAlign: "left",
+  },
+  subtitle: {
+    color: "#d2d2d2",
+    fontWeight: 400,
+    fontSize: 15,
+  },
+  text: {
+    color: theme.palette.text.secondary,
+  },
+  customBox: {
+    display: "-webkit-box",
+    boxOrient: "vertical",
+    lineClamp: 2,
+    wordBreak: "break-all",
+    overflow: "hidden",
+  },
+  icon: {
+    alignSelf: "start",
+    padding: theme.spacing(2),
+    color: theme.palette.primary.main,
+  },
+}));
+
+const Mention = ({ mention }) => {
+  const classes = useStyles();
+  const keyword = "DolphinCorp";
+  const regex = new RegExp(`${keyword}`, "i");
+
+  useEffect(() => {
+    function highlight(title) {
+      title.innerHTML = title.innerHTML.replace(
+        regex,
+        `<span style="color: #536dfe">${keyword}</span>`
+      );
+    }
+    highlight(document.querySelector("#contentTitle"));
```

Comment:
Directly accessing the DOM in React is a major anti-pattern as React uses a virtual DOM to manage updates. You should be able to handle this inside of the component, maybe with state, to avoid interacting with the DOM

Reply:
Hey Dan, basically I tried some ways like "dangerouslySetInnerHTML" or I have to use libraries like react-html-parser. If I manage it inside components, the browser will escape the HTML tags. The other way I thought might work is to slice the title into 3 parts and return a new string with the keyword independently styled. Do you have any suggestions?

Reply:
Hey Dan, did it in the last way, check it out